nucypher/utilities/decorators.py

```diff
@@ -0,0 +1,65 @@
+import functools
+
+from twisted.logger import Logger
+from typing import Callable
+import inspect
+import eth_utils
+
+
+def validate_checksum_address(func: Callable) -> Callable:
```

Comment:
Wow, that's a cool way of doing it!

Reply:
Yeah - I'm pretty happy about this decorator making its way into the codebase.